File: src/main/java/ru/mousecray/endmagic/runes/RunePart.java
package ru.mousecray.endmagic.runes;

public class RunePart {
    public final int x, y;
    public final EnumPartType type;

    public RunePart(int x, int y, EnumPartType type) {
        this.x = x;
        this.y = y;
        this.type = type;
    }

    public enum EnumPartType {
        FullPixel
    }
}

File: src/main/scala/ru/mousecray/endmagic/runes/RunePart.scala
package ru.mousecray.endmagic.runes

case class RunePart(partType: EnumPartType, colorId: Byte) {

}

object RunePart {

  object EmptyPart extends RunePart(EnumPartType.Empty, 0)

  def apply(partType: EnumPartType, colorId: Int): RunePart = new RunePart(partType, colorId.toByte)

}
